### ISSUE TITLE
Compilation fails on Lubuntu 17.10

### ISSUE THREAD

**Issue by waheedejaz** [objectcomputing/OpenDDS#782]
OpenDDS VERSION: 3.12
    TAO VERSION: 2.2a
    Ubuntu 17.10 running as VM on Windows 10 using VMware Player.

    CONTENTS OF $ACE_ROOT/ace/config.h: 
    [#include "ace/config-linux.h"]

    CONTENTS OF $ACE_ROOT/include/makeinclude/platform_macros.GNU:
    [optimize = 0
     include $(ACE_ROOT)/include/makeinclude/platform_linux.GNU

    CONTENTS OF $ACE_ROOT/bin/MakeProjectCreator/config/default.features
    [no_cxx11=0]

    LEVEL OF URGENCY (HIGH):

    configure script ran successfully. It downloaded ACE_TAO.
    make fails. ACE compilation fails:

```
waheed@waheed-virtual-machine:~/OpenDDS$ make
make[1]: Entering directory '/home/waheed/OpenDDS/ACE_wrappers/ace'

GNUmakefile: /home/waheed/OpenDDS/ACE_wrappers/ace/GNUmakefile.ACE MAKEFLAGS=w

g++ -Wno-deprecated-declarations -ggdb -pthread -fno-strict-aliasing -Wall -W -Wpointer-arith -pipe -D_GNU_SOURCE -DACE_HAS_CUSTOM_EXPORT_MACROS=0   -I/home/waheed/OpenDDS/ACE_wrappers -D__ACE_INLINE__ -I.. -DACE_BUILD_DLL  -c -fPIC -o .shobj/Local_Name_Space.o Local_Name_Space.cpp
In file included from /home/waheed/OpenDDS/ACE_wrappers/ace/OS_NS_Thread.h:32:0,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/Thread_Mutex.h:31,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/Log_Category.h:20,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/Malloc.h:24,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/Malloc_T.h:18,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/Local_Name_Space.h:20,
                 from Local_Name_Space.cpp:3:
/home/waheed/OpenDDS/ACE_wrappers/ace/os_include/os_sched.h:47:6: error: conflicting declaration ‘typedef struct cpu_set_t cpu_set_t’
    } cpu_set_t;
      ^~~~~~~~~
In file included from /usr/include/sched.h:44:0,
                 from /usr/include/pthread.h:23,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/os_include/os_pthread.h:54,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/OS_NS_Thread.h:31,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/Thread_Mutex.h:31,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/Log_Category.h:20,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/Malloc.h:24,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/Malloc_T.h:18,
                 from /home/waheed/OpenDDS/ACE_wrappers/ace/Local_Name_Space.h:20,
                 from Local_Name_Space.cpp:3:
/usr/include/i386-linux-gnu/bits/cpu-set.h:42:3: note: previous declaration as ‘typedef struct cpu_set_t cpu_set_t’
 } cpu_set_t;
   ^~~~~~~~~
/home/waheed/OpenDDS/ACE_wrappers/include/makeinclude/rules.local.GNU:188: recipe for target '.shobj/Local_Name_Space.o' failed
make[1]: *** [.shobj/Local_Name_Space.o] Error 1
make[1]: Leaving directory '/home/waheed/OpenDDS/ACE_wrappers/ace'
GNUmakefile.dist:453: recipe for target 'ACE' failed
make: *** [ACE] Error 2
```

**Comment by iguessthislldo** [comment]
Currently this is an issue with OpenDDS' default version version of ACE with at least some recent versions of Linux. Try adding `--doc_group` to the configure, this fixed this same issue when I first built OpenDDS by downloading an alternative version of ACE.

**Comment by waheedejaz** [comment]
> Try adding --doc_group to the configure,
Tried that, it didn't work for me. 

Anyway, I switched to Ubuntu_16.04 LTS (32 bit) and it compiles and run fine.

**Comment by mitza-oci** [comment]
You can make a 1-line change to the impacted ACE file to get it to compile:
https://github.com/DOCGroup/ACE_TAO/commit/65a6d0bb86e9df6fd323f646a512b9527ec73cb4#diff-4406e39ab8f277144bf49af61fa035d1

This will be included in future OCITAO 2.2a releases.
